cpp/src/arrow/extension/tensor_extension_array_test.cc

```diff
@@ -165,47 +170,51 @@ TEST_F(TestExtensionType, CreateFromArray) {
   ASSERT_EQ(ext_arr->null_count(), 0);
 }
 
+template <typename T>
```

Comment:
No need for this if we just cast to `ExtensionType` below.

Reply:
Indeed. Removing.